src/components/workflowEditor/nodes/CDNode.tsx

```diff
@@ -2,9 +2,33 @@ import React, { Component } from 'react'
 import { Link } from 'react-router-dom'
 import { ReactComponent as Add } from '../../../assets/icons/ic-add.svg'
 import Tippy from '@tippyjs/react'
-import { CDNodeProps } from '../types'
+import { CDNodeProps, CDNodeState } from '../types'
+import { toast } from 'react-toastify'
+import {BUTTON_TEXT, CONFIRMATION_DIALOG_MESSAGING, ERR_MESSAGE_ARGOCD, VIEW_DELETION_STATUS} from '../../../config/constantMessaging'
+import { ConfirmationDialog } from '../../common'
+import warningIconSrc from '../../../assets/icons/info-filled.svg'
+
+
+export class CDNode extends Component<CDNodeProps, CDNodeState> {
+    constructor(props) {
+        super(props)
+        this.state = {
+            showDeletePipelinePopup: false,
+            previousEnvironmentName: '',
+        }
+    }
+
+    onClickShowDeletePipelinePopup = () => {
+        this.setState({
+            showDeletePipelinePopup: true,
+            previousEnvironmentName: this.props.environmentName,
```

Comment:
If we're using environmentName val from prop can we remove the state?

Reply:
Done